src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -146,6 +148,9 @@ public VisualStudioWorkspaceImpl(ExportProvider exportProvider, IAsyncServicePro
                     exportProvider.GetExportedValue<IDiagnosticUpdateSourceRegistrationService>(),
                     exportProvider.GetExportedValue<IAsynchronousOperationListenerProvider>(),
                     _threadingContext), isThreadSafe: true);
+
+            _lspRegistrationSerivce = exportProvider.GetExportedValue<ILspWorkspaceRegistrationService>();
+            _lspRegistrationSerivce.Register(this);
```

Comment:
I don't quite like the pattern where a constructor registers the instance as a side-effect to some service. It couples the type to the service.

We have a concept of `IEventListener<object>`/`IEventListenerStoppable` that performs actions when workspace is created/disposed. See e..
`VisualStudioWorkspaceServiceHubConnector`. Perhaps that could be used for the registration.